chromium_src/chrome/android/java/src/org/chromium/chrome/browser/ntp/IncognitoNewTabPageView.java

```diff
@@ -0,0 +1,267 @@
+                mManager.loadIncognitoLearnMore();
+            }
+        });
+
+        mWidthDp = getContext().getResources().getConfiguration().screenWidthDp;
+        mHeightDp = getContext().getResources().getConfiguration().screenHeightDp;
+
+        adjustView();
+    }
+
+    /**
+     * Initialize the incognito New Tab Page.
+     * @param manager The manager that handles external dependencies of the view.
+     */
+    void initialize(IncognitoNewTabPageManager manager) {
+        mManager = manager;
+    }
+
+    /** @return The IncognitoNewTabPageManager associated with this IncognitoNewTabPageView. */
+    protected IncognitoNewTabPageManager getManager() {
+        return mManager;
+    }
+
+    /**
+     * @see org.chromium.chrome.browser.compositor.layouts.content.
+     *         InvalidationAwareThumbnailProvider#shouldCaptureThumbnail()
+     */
+    boolean shouldCaptureThumbnail() {
+        if (getWidth() == 0 || getHeight() == 0) return false;
+
+        return getWidth() != mSnapshotWidth
+                || getHeight() != mSnapshotHeight
+                || mScrollView.getScrollY() != mSnapshotScrollY;
+    }
+
+    /**
+     * @see org.chromium.chrome.browser.compositor.layouts.content.
+     *         InvalidationAwareThumbnailProvider#captureThumbnail(Canvas)
+     */
+    void captureThumbnail(Canvas canvas) {
+        ViewUtils.captureBitmap(this, canvas);
+        mSnapshotWidth = getWidth();
+        mSnapshotHeight = getHeight();
+        mSnapshotScrollY = mScrollView.getScrollY();
+    }
+
+    @Override
+    public void onAttachedToWindow() {
+        super.onAttachedToWindow();
+        assert mManager != null;
+        if (mFirstShow) {
+            mManager.onLoadingComplete();
+            mFirstShow = false;
+        }
+    }
+
+    @Override
+    public void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        // View#onConfigurationChanged() doesn't get called when resizing this view in
+        // multi-window mode, so #onMeasure() is used instead.
+        Configuration config = getContext().getResources().getConfiguration();
+        if (mWidthDp != config.screenWidthDp || mHeightDp != config.screenHeightDp) {
+            mWidthDp = config.screenWidthDp;
+            mHeightDp = config.screenHeightDp;
+            adjustView();
+        }
+
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+    }
+
+    private void adjustView() {
+        adjustIcon();
+        adjustLayout();
+        adjustLearnMore();
+    }
+
+    /**
+     * @param element Resource ID of the element to be populated with the bulletpoints.
+     * @param content String ID to serve as the text of |element|. Must contain an <em></em> span,
+     *         which will be emphasized, and three <li> items, which will be converted to
+     *         bulletpoints.
+     * Populates |element| with |content|.
+     */
+    private void populateBulletpoints(@IdRes int element, @StringRes int content) {
```

Comment:
I don't see this function being called anywhere, unless I'm missing it...

Reply:
k, removed it.